### ISSUE TITLE
MEME.bf error for partition

### ISSUE THREAD

**Issue by sjspielman** [veg/hyphy#545]
MEME fails to complete when a partition (in the menu prompt) is specified. Dataset to reproduce error is attached. [test.fasta.txt](https://github.com/veg/hyphy/files/979054/test.fasta.txt). Note that this error does not occur in FEL.bf.

Stdout w/ error:

```
### Improving branch lengths, nucleotide substitution biases, and global dN/dS ratios under a full codon model
* Log(L) =  -144.01
* non-synonymous/synonymous rate ratio =   0.8543
Error:
Attempting to operate on an undefined value; this is probably the result of an earlier 'soft' error condition Exiting...

Function call stack
1 : Branch under condition 'Eval(condition)'
	to
		utility.Filter.return_object[utility.Filter.keys[utility.Filter.k]]=^(lambda_name)
	else
		utility.Filter.k+=1
-------
2 : result_row[7]=utility.Array1D(utility.Filter(result["ebf"],"_value_","_value_>=100"))
-------
3 : Call(result_callback,0,Eval(job+'('+Join(",",utility.Map(arguments,"_value_","utility.convertToArgumentString (_value_)"))+')'),arguments)
-------
4 : mpi.QueueJob(meme.queue,"meme.handle_a_site",{"0":"meme.site_likelihood","1":"meme.site_likelihood_bsrel","2":alignments.serialize_site_filter((meme.filter_specification[meme.partition_index])["name"],(_pattern_info_["sites"])[0]),"3":meme.partition_index,"4":_pattern_info_,"5":meme.site_model_mapping},"meme.store_results")
-------
5 : ExecuteCommands in string transform using basepath .
-------
6 : utility.ForEachPair(meme.site_patterns,"_pattern_","_pattern_info_",'
            if (_pattern_info_["is_constant"]) {
                meme.store_results (-1,None,{"0" : "meme.site_likelihood",
                							 "1" : "meme.site_likelihood_bsrel",
											 "2" : None,
											 "3" : meme.partition_index,
											 "4" : _pattern_info_,
											 "5" : meme.site_model_mapping
                                     });
            } else {
                mpi.QueueJob (meme.queue, "meme.handle_a_site", {"0" : "meme.site_likelihood",
                												 "1" : "meme.site_likelihood_bsrel",
                                                                 "2" : alignments.serialize_site_filter
                                                                   ((meme.filter_specification[meme.partition_index])["name"],
                                                                   (_pattern_info_["sites"])[0]),
																 "3" : meme.partition_index,
																 "4" : _pattern_info_,
																 "5" : meme.site_model_mapping
																	},
																	"meme.store_results");
            }
        ')
-------
```

**Comment by stevenweaver** [comment]
Dear @sjspielman ,

In the future, can you please provide which environment you are using when reporting issues?

Best,
Steven

**Comment by sjspielman** [comment]
Error seen on Mac Yosemite in v2.3-dev.

**Comment by stevenweaver** [comment]
This is an issue with batch language incorrectly parsing scientific notation. I should have a fix shortly.

**Comment by stevenweaver** [comment]
merged into v2.3-dev
